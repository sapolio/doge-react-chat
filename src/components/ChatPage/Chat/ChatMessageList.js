import React from 'react';
import { withStyles } from 'material-ui/styles';
import ChatMessage from './ChatMessage';
import Typography from 'material-ui/Typography';

const styles = theme => ({
  messagesWrapper: {
    overflowX: 'scroll',
    height: '100%',
    width: '100%',
    paddingTop: theme.spacing.unit * 3,
    paddingBottom: '120px',
  },
});

class ChatMessageList extends React.Component {
  componentDidMount() {
    this.scrollDownHistory();
  }

  componentDidUpdate() {
    this.scrollDownHistory();
  }

  scrollDownHistory() {
    const messagesWrapper = this.refs.messagesWrapper;
    if (messagesWrapper) {
      messagesWrapper.scrollTop = messagesWrapper.scrollHeight;
    }
  }

  render() {
    const { classes, messages, activeUser } = this.props;

    return !messages.length ? (
      <Typography variant="display1">
        Not so messages wow
      </Typography>
    ) : (
        <div className={classes.messagesWrapper} ref="messagesWrapper">
          {messages.map((message, index) => (
            <ChatMessage
              key={message._id}
              fromMe={message.sender._id === activeUser._id}
              {...message} />
          )
          )}
        </div>)


  }
}

export default withStyles(styles)(ChatMessageList);
