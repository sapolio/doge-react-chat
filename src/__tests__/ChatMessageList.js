/* eslint-env jest */
import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import ChatMessageList from '../components/ChatPage/Chat/ChatMessageList';

jest.mock('moment', () => () => ({ fromNow: () => '2 days ago' }));

const props = {
  messages: [
    {
      _id: '123456789',
      chatId: '12345',
      content: 'Hello, World!',
      sender: {
        _id: '12345',
        username: 'me',
      },
      createdAt: 'string',
    },
    {
      _id: '321bar123',
      chatId: '12345',
      content: 'Hello, React!',
      sender: {
        _id: '54321',
        username: 'someone',
      },
      createdAt: 'string',
    },
  ],
  activeUser: {
    _id: '12345',
    username: 'me',
    isMember: true,
    isCreator: true,
    isChatMember: true,
  },
};

describe('<ChatMessageList />', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<ChatMessageList {...props} />, div);
    ReactDOM.unmountComponentAtNode(div);
  });
  it('renders no messages', () => {
    const tree = renderer
      .create(<ChatMessageList activeUser={props.activeUser} messages={[]} />)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
  it('renders with messages', () => {
    const tree = renderer.create(<ChatMessageList {...props} />).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
