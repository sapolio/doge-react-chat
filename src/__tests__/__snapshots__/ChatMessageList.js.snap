// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ChatMessageList /> renders no messages 1`] = `
<h1
  className="MuiTypography-root-62 MuiTypography-display1-66"
>
  Not so messages wow
</h1>
`;

exports[`<ChatMessageList /> renders with messages 1`] = `
<div
  className="ChatMessageList-messagesWrapper-61"
>
  <div
    className="ChatMessage-messageWrapper-86 ChatMessage-messageWrappperFromMe-87"
  >
    <div
      className="MuiAvatar-root-91 MuiAvatar-colorDefault-92"
      style={
        Object {
          "backgroundColor": "#ffc107",
        }
      }
    >
      M
    </div>
    <div
      className="MuiPaper-root-94 MuiPaper-shadow2-98 MuiPaper-rounded-95 ChatMessage-message-89 ChatMessage-messageFromMe-90"
    >
      <span
        className="MuiTypography-root-62 MuiTypography-caption-72"
        style={
          Object {
            "color": "#ffc107",
          }
        }
      >
        me
      </span>
      <p
        className="MuiTypography-root-62 MuiTypography-body1-71"
      >
        Hello, World!
      </p>
      <span
        className="MuiTypography-root-62 MuiTypography-caption-72"
      >
        2 days ago
      </span>
    </div>
  </div>
  <div
    className="ChatMessage-messageWrapper-86"
  >
    <div
      className="MuiAvatar-root-91 MuiAvatar-colorDefault-92"
      style={
        Object {
          "backgroundColor": "#ffc107",
        }
      }
    >
      S
    </div>
    <div
      className="MuiPaper-root-94 MuiPaper-shadow2-98 MuiPaper-rounded-95 ChatMessage-message-89"
    >
      <span
        className="MuiTypography-root-62 MuiTypography-caption-72"
        style={
          Object {
            "color": "#ffc107",
          }
        }
      >
        someone
      </span>
      <p
        className="MuiTypography-root-62 MuiTypography-body1-71"
      >
        Hello, React!
      </p>
      <span
        className="MuiTypography-root-62 MuiTypography-caption-72"
      >
        2 days ago
      </span>
    </div>
  </div>
</div>
`;
